lib/components/user/places/favorite-places-pane.js

```diff
@@ -0,0 +1,60 @@
+import { FieldArray } from 'formik'
+import React, { Component } from 'react'
+import { ControlLabel } from 'react-bootstrap'
+
+import { isHomeOrWork } from '../../../util/user'
+import FavoritePlaceRow from './favorite-place-row'
+
+/**
+ * Renders an editable list user's favorite locations, and lets the user add a new one.
+ */
+class FavoriteLocationsPane extends Component {
+  _handleDelete = (arrayHelpers, place, index) =>
+    () => {
+      const isFixed = isHomeOrWork(place)
+      if (isFixed) {
+        // Just clear the address if place is fixed.
+        const newPlace = {
+          ...place,
+          address: ''
+        }
+        arrayHelpers.replace(index, newPlace)
+      } else {
+        arrayHelpers.remove(index)
+      }
+    }
+
+  render () {
+    const { values: userData } = this.props
+    const { savedLocations } = userData
+
+    return (
+      <div>
+        <ControlLabel>Add the places you frequent often to save time planning trips:</ControlLabel>
+
+        <FieldArray
+          name='savedLocations'
+          render={arrayHelpers => (
+            <>
+              {savedLocations.map((place, index) => (
+                <FavoritePlaceRow
+                  isFixed={isHomeOrWork(place)}
+                  key={index}
+                  onDelete={this._handleDelete(arrayHelpers, place, index)}
```

Comment:
This doesn't actually delete the place/persist the user data. So, when a user clicks delete, and then clicks the item to add a new location, the old location has re-appeared. I think this should be fixed (even before we address the potential removal of the Save/Cancel buttons at the bottom of the user settings page).

Reply:
This is updated in https://github.com/opentripplanner/otp-react-redux/pull/280/commits/c08040eee92a71d5dd0ca253f98e18bddff97112.